packages/core/src/Stores/client-store.js

```diff
@@ -377,6 +377,10 @@ export default class ClientStore extends BaseStore {
         // TODO revert this when all landing companies are accepted.
         // return 'mt_financial_company' in this.landing_companies || 'mt_gaming_company' in this.landing_companies;
         if ('mt_financial_company' in this.landing_companies || 'mt_gaming_company' in this.landing_companies) {
+            // TODO: [deriv-eu] Update this when all EU functionalities merge into production.
+            if (this.root_store.ui.is_eu_enabled) {
+                return true;
+            }
```

Comment:
This won't work in all cases since we need to rely on `mt_financial_company` and `mt_gaming_company` anyway, 
So I suggest we refactor this into: 
```js
if (this.root_store.ui.is_eu_enabled) return 'mt_financial_company' in this.landing_companies || 'mt_gaming_company' in this.landing_companies;
```